app/mci/pkg/gcp/backendservice/backendservicesyncer.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/api/compute/v1"
+	ingressbs "k8s.io/ingress-gce/pkg/backends"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/healthcheck"
+	utilsnamer "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/namer"
+	sp "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/serviceport"
+)
+
+type BackendServiceSyncer struct {
+	namer *utilsnamer.Namer
+	bsp   ingressbs.BackendServices
+}
+
+func NewBackendServiceSyncer(namer *utilsnamer.Namer, bsp ingressbs.BackendServices) BackendServiceSyncerInterface {
+	return &BackendServiceSyncer{
+		namer: namer,
+		bsp:   bsp,
+	}
+}
+
+// Ensure this implements BackendServiceSyncerInterface.
+var _ BackendServiceSyncerInterface = &BackendServiceSyncer{}
+
+// EnsureBackendService ensures that the required backend service exists.
+// Does nothing if it exists already, else creates a new one.
+func (h *BackendServiceSyncer) EnsureBackendService(lbName string, ports []sp.ServicePort, hcMap healthcheck.HealthChecksMap, npMap NamedPortsMap, igLinks []string) error {
+	fmt.Println("Ensuring backend services")
+	glog.V(5).Infof("Received health checks map: %v", hcMap)
+	glog.V(5).Infof("Received named ports map: %v", npMap)
+	glog.V(5).Infof("Received instance groups: %v", igLinks)
+	var err error
+	for _, p := range ports {
+		if hcErr := h.ensureBackendService(lbName, p, hcMap[p.Port], npMap[p.Port], igLinks); hcErr != nil {
+			hcErr = fmt.Errorf("Error %s in ensuring backend service for port %v", hcErr, p)
+			// Try ensuring backend services for all ports and return all errors at once.
+			err = multierror.Append(err, hcErr)
+		}
+	}
+	return err
+}
+
+func (h *BackendServiceSyncer) ensureBackendService(lbName string, port sp.ServicePort, hc *compute.HealthCheck, np *compute.NamedPort, igLinks []string) error {
+	fmt.Println("Ensuring backend service for port:", port)
+	if hc == nil {
+		return fmt.Errorf("missing health check")
+	}
+	if hc.SelfLink == "" {
+		glog.V(2).Infof("Unexpected: empty self link in hc: %v", hc)
+		return fmt.Errorf("missing self link in health check %s", hc.Name)
+	}
+	if np == nil {
+		return fmt.Errorf("missing corresponding named port on the instance group")
+	}
+	desiredBS := h.desiredBackendService(lbName, port, hc.SelfLink, np.Name, igLinks)
+	name := desiredBS.Name
+	// Check if backend service already exists.
+	existingBS, err := h.bsp.GetGlobalBackendService(name)
+	if err == nil {
+		fmt.Println("Backend service", name, "exists already. Checking if it matches our desired backend service", name)
+		glog.V(5).Infof("Existing backend service: %v\n, desired backend service: %v", existingBS, *desiredBS)
+		// Backend service with that name exists already. Check if it matches what we want.
+		if backendServiceMatches(desiredBS, existingBS) {
+			// Nothing to do. Desired backend service exists already.
+			fmt.Println("Desired backend service exists already")
```

Comment:
why not use glog?

Reply:
glog also prints the timestamp, filename and line number. It is great for debug logging but not for user visible messages.
kubectl also uses fmt for user messages and glog for debugging messages.